packages/jw-player/src/mixins/ChapterMixin.js

```diff
@@ -0,0 +1,127 @@
+import { css, html, render } from 'lit';
+
+// eslint-disable-next-line func-names
+const chapterNavigationTemplate = function (chapters) {
+  return html`
+    <a class="close" @click=${this.__toggleChapterNavigation.bind(this)}>&#10005;</a>
+    <ul class="chapter-navigation-list">
+      ${chapters.map(
+        (chapter, index) =>
+          html`
+            <li data-index=${index} @click=${this.__chapterSeek.bind(this)}>
+              <a>${chapter.data.text}</a>
+            </li>
+          `
+      )}
+    </ul>
+  `;
+};
+
+export function ChapterMixin(BaseClass) {
+  class Mixin extends BaseClass {
+    __path = 'https://cxl.com/institute/wp-content/plugins/cxl-jwplayer/';
+
+    async __createChapterNavigation() {
+      const chapters = await this.__getChapters();
+
+      this.__chapterNavigation = document.createElement('div');
+      this.__chapterNavigation.classList.add('chapter-navigation');
+      this.__chapterNavigation.hidden = true;
+
+      render(chapterNavigationTemplate.bind(this)(chapters), this.__chapterNavigation);
+
+      this.__jwPlayerContainer.appendChild(this.__chapterNavigation);
+
+      this.__jwPlayer.addButton(
+        `${this.__path}images/chapter-bookmark-icon.svg`,
+        'Show Chapters',
+        this.__toggleChapterNavigation.bind(this),
+        'toggle-chapters'
+      );
+    }
+
+    __addStyle() {
+      const style = document.createElement('style');
+      render(
+        css`
```

Comment:
Inline css? can't be loaded from external file? Similar to how `cxl-ui` components load css?

Reply:
I refactored the styles into a separate file.